src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1827,9 +1827,18 @@ def generate(
         inputs_embeds = self.get_input_embeddings()(input_ids)
         inputs_embeds = torch.cat([language_model_inputs, inputs_embeds.to(language_model_inputs.device)], dim=1)
 
+        if not self.language_model.config.is_encoder_decoder:
```

Comment:
Is this to ensure the generation output keeps the expected max_length/min_length? If so, let's add a comment!

Reply:
yep, will do so